### ISSUE TITLE
Selective preset apply

### ISSUE THREAD

**Issue by travnick** [darktable-org/darktable#17111]
**Is your feature request related to a problem? Please describe.**

We do create presets, to automatically apply some work for us. Unfortunately, I see no option to apply presets partially.

It's quite visible "lens correction" module. I've created preset for my lens (about to fix TCA only) and then, when applied, it uses settings from my previous shoot, so overrides: `mm`, `f`, `d`, camera model etc.

It's quite useless then, since I have to perform my steps manually anyway: hit reset (to automatically determine parameters), and choose TCA only.

The case is also for other automatic modules, like `filmc rgb`, there are some defaults applied, but in most cases the first try is to use "auto-tune levels". It would be nice to have an option to define such behaviour.

Currently, presets are only usable with modules where it's required to have **all** module parameters predefined and used.

**Describe the solution you'd like**

I'd like to have presets with options (per parameter) like `use my hardcoded value`, `use default` and `use auto/from metadata`.

Without that, presets are not usable for all modules, and may introduce unwanted changes to the image.

**Comment by ralfbrown** [comment]
Currently, presets are all-or-nothing due to internal design - they replace the module's entire parameter block with the values stored in the preset.  Partial application isn't all that hard to implement internally, but a usable UI would be a real headache to design and implement.

**Comment by wpferguson** [comment]
This issue can be solved using a lua script. Lua has access to all the image information, darktable state information, and allows you to implement complex logic and control the UI.

A lua script could:
* get triggered when an image is loaded into darkroom
* check the lens correction module to see if it's on
* turn it on if necessary
* depending on the logic, set the correction to TCA only

Your logic can be as complex as necessary

**Comment by jenshannoschwalm** [comment]
We might want enable cropping here automatically for unsupported cameras?

**Comment by AndrewKeYanzhe** [comment]
@wpferguson could you point me towards how to create a lua script that enables geometric correction, but not TCA and vignette?

Thank you. Your help is much appreciated :)

**Comment by wpferguson** [comment]
@AndrewKeYanzhe yes, I can give you some pointers.  Give me a couple of days, I have a big dose of real life to deal with.

**Comment by wpferguson** [comment]
@AndrewKeYanzhe I created a repository, https://github.com/wpferguson/darktable-lua-scripting-aids, which contains a script template and snippets of code to use.  I'll also attach a script that runs in darkroom when you load an image and automatically straightens it.  You can replace the straighten code and interaction with `rotate and perspective`  with interactions with `lens correction`.  You can go into the shortcuts preferences and find out the correct calls to rotate and perspective by hovering the various parts.

[auto_straighten.zip](https://github.com/user-attachments/files/23010850/auto_straighten.zip)
